tests/testthat/test_penaltyLearning.R

```diff
@@ -1,11 +1,141 @@
   n.zero <- sum(l1.norm.vec == 0)
   expect_equal(n.zero, 1)
 })
+
+pred.loglik <- function(pred.mean, pred.scale, target.mat){
+  stopifnot(identical(ncol(target.mat), 2L))
+  n.obs <- nrow(target.mat)
+  stopifnot(identical(nrow(pred.mean), n.obs))
+  stopifnot(identical(nrow(pred.scale), n.obs))
+  prob.below.left <- pnorm(target.mat[,1], pred.mean, pred.scale)
+  prob.below.right <- pnorm(target.mat[,2], pred.mean, pred.scale)
+  log(prob.below.right-prob.below.left)
+}
+
+set.error.list <- list()
+for(set.name in names(sets)){
+  is.set <- sets[[set.name]]
+  pred.mat <- predict(fit, penalty.learning$X.mat[is.set,])
+  y.set <- penalty.learning$y.mat[is.set,]
+  too.lo <- colSums(pred.mat < y.set[,1])
+  too.hi <- colSums(y.set[,2] < pred.mat)
+  errors <- too.lo + too.hi
+  scale.mat <- matrix(fit$scale, nrow(y.set), length(fit$scale), byrow=TRUE)
+  log.prob <- pred.loglik(pred.mat, scale.mat, y.set)
+  set.error.list[[set.name]] <- data.frame(
+    set.name,
+    lambda=fit$lambda,
+    too.lo,
+    too.hi,
+    errors,
+    iterations=fit$n_iters,
+    surrogate.loss=-colMeans(log.prob),
+    percent.error=100*errors/nrow(y.set))
+}
+set.error <- do.call(rbind, set.error.list)
+train.error <- subset(set.error, set.name=="train")
+
+if(interactive()){
+  library(ggplot2)
+
+  tidy <- tidydf(fit)
+  some <- subset(tidy, variable != "(Intercept)" & arclength < 2)
+
+  ggplot()+
+    geom_path(aes(-log(lambda), weight, color=variable, group=variable),
+              data=some)
+
+  ggplot()+
+    geom_path(aes(arclength, weight, color=variable, group=variable),
+              data=some)+
+    geom_point(aes(arclength, weight, color=variable, group=variable),
+               shape=1,
+               data=subset(some, weight != 0))
+
+  ggplot()+
+    theme_bw()+
+    theme(panel.margin=grid::unit(0, "lines"))+
+    facet_grid(metric ~ ., scales="free")+
+    geom_path(aes(-log(lambda), surrogate.loss, color=set.name, group=set.name),
+              data=data.frame(set.error, metric="surrogate loss"))+
+    geom_path(aes(-log(lambda), iterations),
+              data=data.frame(train.error, metric="iterations"))+
+    geom_path(aes(-log(lambda), percent.error, color=set.name, group=set.name),
+              data=data.frame(set.error, metric="percent incorrect intervals"))
+
+  plot(log(fit$lambda))
+
+}
+
+test_that("train set surrogate loss always decreases", {
+  diff.vec <- diff(train.error$surrogate.loss)
+  expect_true(all(diff.vec < 0))
+})
+
+test_that("error_status is 0", {
+  expect_equal(fit$error_status, 0)
+})
+
+## Another fit with standardize=FALSE.
+X.unscaled <- X.train
+mean.vec <- colMeans(X.unscaled)
+M <- matrix(
+  mean.vec, nrow(X.unscaled), ncol(X.unscaled), byrow=TRUE)
+X.centered <- X.unscaled - M
+sd.vec <- sqrt(colSums(X.centered * X.centered)/nrow(X.centered))
+S <- diag(1/sd.vec)
+X.scaled <- X.centered %*% S
+##X.scaled <- X.unscaled %*% S #UNCOMMENT TO MAKE TESTS BELOW PASS.
```

Comment:
uncomment this line and the other one below, and then we use scaling but not centering, and the tests below pass.